app/assets/javascripts/views/schedule.js

```diff
@@ -53,6 +56,17 @@ Yacs.views.schedule = function (data) {
     crnListElement.textContent = 'CRNs: ' + scheduleData.crns.join(', ');
   }
 
+  /* binds event if user has any section selected. Needed when clear btn on schedule page */
+  if(Yacs.user.getSelections().length != 0){
+    Yacs.on('click', clearSwitchElement, function () {
+      /* TODO comment the line if using header button */
+      console.log("clearing selections...");
+      Yacs.user.clearSelections();
+      var event = document.createEvent('Event');
+      event.initEvent('click',true,true);
+      document.querySelector("#schedule-btn").dispatchEvent(event);
```

Comment:
For now I just copy & pasted the code to generate the schedule page from yacs.js